pkg/controller/workload/job/job_controller.go

```diff
@@ -519,7 +520,7 @@ func ConstructWorkloadFor(ctx context.Context, client client.Client,
 	job *batchv1.Job, scheme *runtime.Scheme) (*kueue.Workload, error) {
 	w := &kueue.Workload{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      job.Name,
+			Name:      GetWorkloadNameForJob(job.Name),
```

Comment:
Kind of, workloads are generally API objects that should be "technical details" to the users. But, it might be worth mentioning, I will ask to add the note when we prepare release notes for 0.3.0.